shared/livemark-store.js

```diff
@@ -1,163 +1,39 @@
 "use strict";
 
-/**
- * Asynchronous ES6 Map that syncs with WebExtension storage
- */
-class StoredMap extends Map {
-  /**
-   * Build a StoredMap
-   * @param storageKey The key that will be use to save and retreive the map.
-   */
-  constructor(storageKey) {
-    const getFromStorage = async () => {
-      const found = await browser.storage.local.get(storageKey);
-      if (!found.hasOwnProperty(storageKey)) {
-        return [];
-      }
-
-      return found[storageKey];
-    };
-
-    const setupSync = () => {
-      browser.storage.onChanged.addListener(async (changes, area) => {
-        if (area !== "local" || !changes.hasOwnProperty(storageKey)) {
-          return;
-        }
-        const objectEquals = (a, b) => {
-          for (const key in a) {
-            if (!(key in b) || a[key] !== b[key]) {
-              return false;
-            }
-          }
-          for (const key in b) {
-            if (!(key in a) || a[key] !== b[key]) {
-              return false;
-            }
-          }
-          return true;
-        };
-
-        const deleted = [];
-        const changed = [];
-        const stored = new Map(changes[storageKey].newValue);
-        for (const key of this.keys()) {
-          if (!stored.has(key)) {
-            deleted.push(key);
-          }
-        }
-
-        for (const [key, value] of stored) {
-          if (!this.has(key) || !objectEquals(value, this.get(key))) {
-            changed.push(key);
-          }
-        }
-
-        for (const deletion of deleted) {
-          super.delete(deletion);
-        }
+const PREFIX = "livemark.";
```

Comment:
very minor nit: can we use the plural form `livemarks.` ? Just so we're consistent with the `settings.` I like using the prefs naming scheme :)